### PR TITLE
Sync upstream prometheus

### PR DESCRIPTION
**Pull request by pracucci**
Synching upstream prometheus because I want to get https://github.com/prometheus/prometheus/pull/13843

What's in this sync:
- https://github.com/prometheus/prometheus/pull/13843
- https://github.com/prometheus/prometheus/pull/12503
- https://github.com/prometheus/prometheus/pull/14396

### CODE REVIEW

File: promql/parser/parse.go [hunk 1]
```diff
@@ -447,6 +447,10 @@ func (p *parser) newAggregateExpr(op Item, modifier, args Node) (ret *AggregateE
 
 	desiredArgs := 1
 	if ret.Op.IsAggregatorWithParam() {
+		if !EnableExperimentalFunctions && (ret.Op == LIMITK || ret.Op == LIMIT_RATIO) {
+			p.addParseErrf(ret.PositionRange(), "limitk() and limit_ratio() are experimental and must be enabled with --enable-feature=promql-experimental-functions")
```
Comment (colega) [inline promql/parser/parse.go:451]: Will we have this feature enabled in Mimir? If not, we should modify this message (maybe make it configurable?)

**Review by colega — APPROVED** [review]
LGTM, left one minor comment.

File: promql/parser/parse.go (hunk 1 above)
Comment (pracucci) [reply promql/parser/parse.go]: Good point. I just customised the message, adding a comment to explain why (in case of future conflicts to resolve).

Comment (colega) [reply promql/parser/parse.go]: Thank you, lgtm.

**Pull request merged**